R/plotpermutation.R

```diff
@@ -30,27 +32,40 @@ plotpermutation <- function(x){
   aes <- ggplot2::aes
   element_blank <- ggplot2::element_blank
 
-  p<-ggplot2::ggplot(data=plotdata[plotdata$type=="Permutation", ], aes(x=x, y=y, group=label), aspect="iso") +
-    ggplot2::scale_colour_gradientn("log(p-val)",
+  p<-ggplot2::ggplot(data=plotdata[plotdata$type=="Permutation", ], aes(x=x, y=y, group=label), aspect="iso")
+  if (coloring %in% c("p","p.adj","fish")) {
+    p <- p +
+      ggplot2::scale_colour_gradientn("log(p-val)",
         colours=palette$grpalette,
         values = palette$values,
         limits = palette$limits
-      ) +
-      ggplot2::geom_path(color = "Grey80") +
-      ggplot2::geom_path(data=plotdata[plotdata$type=="Actual", ], aes(x=x, y=y, group=label, color = cc.10))+
-      ggplot2::geom_point(data=plotdata[plotdata$type=="Actual", ], aes(x=x, y=y, group=label, color = cc.10), size=2) +
-      ggplot2::theme_bw() +
-      ggplot2::theme(axis.title.x = element_blank(),
-            axis.text.x = element_blank(),
-            axis.ticks.x = element_blank(),
-            axis.title.y = element_blank(),
-            axis.text.y = element_blank(),
-            axis.ticks.y = element_blank()) +
-      ggplot2::theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
-      ggplot2::labs(color = "Type") +
-      ggplot2::coord_fixed(ratio=1) +
-      ggplot2::ggtitle(Title) +
-      ggplot2::annotate(geom="text", x=data2.labels$xpos, y=data2.labels$ypos,
+      )
+  } else if (coloring != "Terms") {
+    p <- p + ggplot2::scale_colour_gradient(coloring)
+  }
+  p <- p + ggplot2::geom_path(color = "Grey80")
+  if (coloring!="Terms") {
+    p <- p +
```

Comment:
same as before.

Reply:
As before: in one case color appears within the aes brackets and in the other it does not.